### ISSUE TITLE
Bug word frequency protocols (smooth)

### ISSUE THREAD

**Issue by fredrik1984** [welfare-state-analytics/welfare_state_analytics#108]
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/word_trends_gui.py in update_plot(*_)
    125             smoothers = DEFAULT_SMOOTHERS if gui.smooth.value else []
    126 
--> 127             data = gui.current_display.compile(_corpus, indices, smoothers=smoothers)
    128 
    129             state.data = data

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/notebook/word_trends/displayers/data_compilers.py in compile_multiline_data(corpus, indices, **kwargs)
     32         ys_j = bag_term_matrix.getcol(j).A.ravel()
     33         for smoother in smoothers:
---> 34             xs_j, ys_j = smoother(xs_j, ys_j)
     35         xs_data.append(xs_j)
     36         ys_data.append(ys_j)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/penelope/common/curve_fit.py in pchip_spline(xs, ys)
     51     if isinstance(ys, scipy.sparse.spmatrix):
     52         ys = ys.todense().A1
---> 53     spliner = scipy.interpolate.PchipInterpolator(xs, ys)
     54     s_xs = np.arange(xs.min(), xs.max() + 0.1, 0.1)
     55     s_ys = spliner(s_xs)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in __init__(self, x, y, axis, extrapolate)
    228     """
    229     def __init__(self, x, y, axis=0, extrapolate=None):
--> 230         x, _, y, axis, _ = prepare_input(x, y, axis)
    231         xp = x.reshape((x.shape[0],) + (1,)*(y.ndim-1))
    232         dk = self._find_derivatives(xp, y)

~/Library/Caches/pypoetry/virtualenvs/word-frequency-protocols-0.2.2-2TYASCss-py3.8/lib/python3.8/site-packages/scipy/interpolate/_cubic.py in prepare_input(x, y, axis, dydx)
     46         raise ValueError("`x` must contain at least 2 elements.")
     47     if x.shape[0] != y.shape[axis]:
---> 48         raise ValueError("The length of `y` along `axis`={0} doesn't "
     49                          "match the length of `x`".format(axis))
     50 

ValueError: The length of `y` along `axis`=0 doesn't match the length of `x`
```

**Comment by roger-mahler** [comment]
The plot fails beacuse the data is grouped by document instead of year. A temporary fix is to replace the content in the thirds code cell with the following code:
 
```python
def loaded_callback_wrapper(
    output,
    *,
    corpus=None,
    corpus_folder: str = None,
    corpus_tag: str = None,
    n_count: int = 10000,
    **kwargs,
):
    corpus = corpus.group_by_year()
    loaded_callback(
        output, corpus=corpus, corpus_folder=corpus_folder, corpus_tag=corpus_tag, n_count=n_count, **kwargs
    )

vectorize_corpus_gui.display_gui('*sparv4.csv.zip', generated_callback=loaded_callback_wrapper)

```

**Comment by roger-mahler** [comment]
Resolved in v0.2.3.
